core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -191,6 +192,8 @@
 
     private final IndexShardOperationCounter indexShardOperationCounter;
 
+    private EnumSet<IndexShardState> readAllowedStates = EnumSet.of(IndexShardState.STARTED, IndexShardState.RELOCATED, IndexShardState.POST_RECOVERY);
```

Comment:
great. Can we do the same to the write allowed states? writeAllowedOnPrimaryStates, writeAllowedOnReplicaStates?